src/App.tsx

```diff
@@ -1,26 +1,302 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
+import {
+  deleteTodo,
+  getTodos,
+  patchTodo,
+  postTodo,
+  USER_ID,
+} from './api/todos';
+import { Filters } from './constants/filter';
+import { TodoList } from './components/TodoList';
+import { NewTodo } from './components/NewTodo';
+import { Todo } from './types/Todo';
+import { TodoComp } from './components/TodoComp';
+import { Filter } from './components/Filter';
 
-const USER_ID = 0;
+const getPreparedTodos = (todos: Todo[], filter: Filters) => {
+  const newTodos = [...todos];
+
+  switch (filter) {
+    case Filters.Active:
+      return newTodos.filter(todo => todo.completed === false);
+    case Filters.Completed:
+      return newTodos.filter(todo => todo.completed === true);
+    default:
+      return newTodos;
+  }
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string>('');
+  const [filter, setFilter] = useState<Filters>(Filters.All);
+  const [loading, setLoading] = useState(true);
```

Comment:
```suggestion
  const [isLoading, setIsLoading] = useState(true);
```

[Tips on naming boolean variables](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)

Reply:
changed the variable name